lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/classloader/ClassLoaderSafeConnectorMetadata.java

```diff
@@ -649,10 +649,18 @@ public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
             ConnectorInsertTableHandle insertHandle,
             Collection<Slice> fragments,
             Collection<ComputedStatistics> computedStatistics,
-            List<ConnectorTableHandle> sourceTableHandles)
-    {
-        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
-            return delegate.finishRefreshMaterializedView(session, tableHandle, insertHandle, fragments, computedStatistics, sourceTableHandles);
+            List<ConnectorTableHandle> sourceTableHandles,
+            List<String> sourcePolymorphicTableFunctions)
```

Comment:
sourcePolymorphicTableFunctions -> sourceTableFunctions 

(the fact that they are polymorphic is irrelevant)